gcp/table_gcp_kms_key_ring.go

```diff
@@ -0,0 +1,175 @@
+			},
+
+			// GCP standard columns
+			{
+				Name:        "location",
+				Description: ColumnDescriptionLocation,
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromP(gcpKmsKeyRingTurbotData, "Location"),
+			},
+			{
+				Name:        "project",
+				Description: ColumnDescriptionProject,
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromP(gcpKmsKeyRingTurbotData, "Project"),
+			},
+		},
+	}
+}
+
+//// LIST FUNCTION
+func listKeyRingDetails(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData) (interface{}, error) {
+	location := plugin.GetMatrixItem(ctx)[matrixKeyLocation].(string)
+	plugin.Logger(ctx).Trace("listKeyRingDetails")
+
+	// Create Service Connection
+	service, err := KMSService(ctx, d)
+	if err != nil {
+		return nil, err
+	}
+
+	// Get project details
+	projectData, err := activeProject(ctx, d)
+	if err != nil {
+		return nil, err
+	}
+	project := projectData.Project
+	resp := service.Projects.Locations.KeyRings.List("projects/" + project + "/locations/" + location)
+	if err := resp.Pages(ctx, func(page *cloudkms.ListKeyRingsResponse) error {
+		for _, ring := range page.KeyRings {
+			d.StreamListItem(ctx, ring)
+		}
+		return nil
+	}); err != nil {
+		return nil, err
+	}
+
+	return nil, nil
+}
+
+//// HYDRATE FUNCTIONS
+
+func getKeyRingDetail(ctx context.Context, d *plugin.QueryData, _ *plugin.HydrateData) (interface{}, error) {
+	matrixLocation := plugin.GetMatrixItem(ctx)[matrixKeyLocation].(string)
+	plugin.Logger(ctx).Trace("getKeyRingDetail")
+
+	// Create Service Connection
+	service, err := KMSService(ctx, d)
+	if err != nil {
+		return nil, err
+	}
+
+	// Get project details
+	projectData, err := activeProject(ctx, d)
+	if err != nil {
+		return nil, err
+	}
+	project := projectData.Project
+
+	name := d.KeyColumnQuals["name"].GetStringValue()
+	location := d.KeyColumnQuals["location"].GetStringValue()
+
+	// to prevent duplicate value
+	if matrixLocation != location {
+		return nil, nil
+	}
+	resp, err := service.Projects.Locations.KeyRings.Get("projects/" + project + "/locations/" + location + "/keyRings/" + name).Do()
+	if err != nil {
+		return nil, err
+	}
+
+	return resp, nil
+}
+
+func getKmsKeyRingIamPolicy(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData) (interface{}, error) {
+	plugin.Logger(ctx).Trace("getKmsKeyRingIamPolicy")
+
+	// Create Service Connection
+	service, err := KMSService(ctx, d)
+	if err != nil {
+		return nil, err
+	}
+
+	keyRing := h.Item.(*cloudkms.KeyRing)
+
+	resp, err := service.Projects.Locations.KeyRings.GetIamPolicy(keyRing.Name).Do()
+	if err != nil {
+		return nil, err
+	}
+
+	return resp, nil
+}
+
+//// TRANSFORM FUNCTIONS
+
+func gcpKmsKeyRingTurbotData(_ context.Context, d *transform.TransformData) (interface{}, error) {
+	key := d.HydrateItem.(*cloudkms.KeyRing)
+	param := d.Param.(string)
+
+	project := strings.Split(key.Name, "/")[1]
+	location := strings.Split(key.Name, "/")[3]
```

Comment:
Can we use the split operation once and use that value?
```suggestion
	data := strings.Split(key.Name, "/")
```